### ISSUE TITLE
Grails 6.1.2 - Application fails to render default page when started with java -Dgrails.env=dev -jar myapp-0.1.war

### ISSUE THREAD

**Issue by klassek65** [grails/grails-core#13415]
### Expected Behavior

The application should start and render the default page

### Actual Behaviour

The application starts... but a request for the default page causes the error :
javax.servlet.ServletException: Could not resolve view with name '/index' in servlet with name 'grailsDispatcherServlet'

### Steps To Reproduce

Using sdkman on Ubuntu 22.04, install java 11.0.22-zulu and grails 6.1.2 and run the following commands in a terminal:
1. grails create-app my.pkg.myapp
2. cd myapp
3. ./gradlew clean bootWar
4. java -Dgrails.env=dev -jar build/libs/myapp-0.1.jar
5. Open a browser to localhost:8080
6. Observe the stacktrace on the terminal: 
klassek@tad-cooper:~/Development/Grails/experimental/myapp$ java -jar -Dgrails.env=dev build/libs/myapp-0.1.war 
Grails application running at http://localhost:8080 in environment: development
2024-03-21 12:23:46.021 ERROR --- [nio-8080-exec-1] .a.c.c.C.[.[.[.[grailsDispatcherServlet] : Servlet.service() for servlet [grailsDispatcherServlet] in context with path [] threw exception [Could not resolve view with name '/index' in servlet with name 'grailsDispatcherServlet'] with root cause

javax.servlet.ServletException: Could not resolve view with name '/index' in servlet with name 'grailsDispatcherServlet'
	at org.springframework.web.servlet.DispatcherServlet.render(DispatcherServlet.java:1385)
        etc.

Interestingly enough...
  ./gradlew bootRun -Dgrails.env=dev
works as expected and renders the default page

This also works as expected and renders the default page
java -Dgrails.env=prod build/libs/myapp-0.1.jar 

### Environment Information

Ubuntu 22.04
Java 11.0.22-zulu

### Version

6.1.2

**Comment by matrei** [comment]
Add `-Dgrails.gsp.view.dir=views` (the value does not matter if you don't have any GSP:s that are not already part of the application. This will trigger the Grails app to use another `GroovyPageResourceLoader` that will find and use the pre-compiled views.

**Comment by guillermocalvo** [comment]
@klassek65 Thanks for your report.

I'd just like to let you know that this specific scenario has never worked for any Grails versions, and currently there's no short-term plan to add support for it. However, you can try the approach suggested by @matrei to work around this limitation.

**Comment by klassek65** [comment]
Hi guillermocalvo and matrei,

Thank you for the work around. I tested the -Dgrails.gsp.view.dir=views with the steps I posted as well as with the addition of a Book domain class and both cases work as you would expect in dev, test, and prod environments.

I am not sure I understand why setting the gps view dir is necessary for running the war file in the dev and test environments but not the prod environment... but I am happy to have a way forward.

Thanks again for looking into this,

klassek65

PS. You might want to consider changing application.yml to include:

```
grails:
    gsp:
      view:
        dir: views

```
as a way of fixing this grails limitation :-)
